### PR TITLE
feat: support tags, refactored & removed default name value for resources. 

### PR DESCRIPTION
**Pull request by vlahane**
## Description

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [x] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Comment by vlahane** [comment]
@dciborow pls review

**Comment by vlahane** [comment]
@dciborow @shenglol you please review and check if this needed, once we have this, I will add diagnostic support for Traffic manager

File: modules/network/traffic-manager/main.bicep [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+  'Geographic'
+  'MultiValue'
+  'Performance'
+  'Priority'
+  'Subnet'
+  'Weighted'
+])
+param trafficRoutingMethod string = 'Performance'
+
+@description('The DNS Time-To-Live (TTL), in seconds.')
```
Comment (shenglol) [inline modules/network/traffic-manager/main.bicep:23]: Mind adding the default value to description?

File: modules/network/traffic-manager/main.bicep [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
+param name string
 
 @description('Relative DNS name for the traffic manager profile, must be globally unique.')
-param trafficManagerDnsName string = 'tmp-${uniqueString(resourceGroup().id, subscription().id, name)}'
+param trafficManagerDnsName string
+
+@description('Tags for the module resources.')
+param tags object = {}
+
+@description('The traffic routing method of the Traffic Manager profile.')
```
Comment (shenglol) [inline modules/network/traffic-manager/main.bicep:12]: Can we add the default value to description?

Comment (vlahane) [reply modules/network/traffic-manager/main.bicep]: done!

File: modules/network/traffic-manager/README.md [hunk 3]
```diff
@@ -17,9 +17,12 @@ The module also allows for the configuration of monitor settings such as protoco
...
-| `prefix`                | `string` | No       | Prefix of Traffic Manager Profile Name                                                                                                                            |
-| `name`                  | `string` | No       | Name of Traffic Manager Profile Resource                                                                                                                          |
-| `trafficManagerDnsName` | `string` | No       | Relative DNS name for the traffic manager profile, must be globally unique.                                                                                       |
+| `name`                  | `string` | Yes      | Name of Traffic Manager Profile Resource                                                                                                                          |
+| `trafficManagerDnsName` | `string` | Yes      | Relative DNS name for the traffic manager profile, must be globally unique.                                                                                       |
```
Comment (shenglol) [inline modules/network/traffic-manager/README.md:21]: Making optional parameters required is a breaking change (see [documentation](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#bumping-major-version)). Can you bump the major version in `version.json` and update the module version in examples?

File: modules/network/traffic-manager/test/main.test.bicep [hunk 4]
```diff
@@ -10,23 +11,34 @@ module file is a deployment test. Make sure at least one test is added.
...
-    endpoints: [{
+    name: 't1${name}'
+    trafficManagerDnsName: 'tmp1-${uniqueString(resourceGroup().id, subscription().id, name)}'
+    endpoints: [ {
         name: 'my-endpoint-1'
         target: 'www.bing.com'
         endpointStatus: 'Enabled'
         endpointLocation: 'eastus'
-    }]
+      } ]
```
Comment (shenglol) [inline modules/network/traffic-manager/test/main.test.bicep:42]: nit: Formatting.

Comment (vlahane) [reply modules/network/traffic-manager/test/main.test.bicep:42]: I think it's bicep fmt did

Comment (shenglol) [reply modules/network/traffic-manager/test/main.test.bicep:42]: Ah, don't worry about it then. That's a weird Bicep formatter behavior, though. I'm going to fix it later.

**Review by shenglol — APPROVED** [review]
:shipit:

File: modules/network/traffic-manager/test/main.test.bicep (hunk 4 above)
Comment (vlahane) [reply modules/network/traffic-manager/test/main.test.bicep:42]: Do we need to raise bug for it ?

**Pull request merged**